libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/coordinator/RouteManager.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.dropin.coordinator
+
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.DropInNavigationViewContext
+import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+
+/**
+ * Class that fetches and sets new route on destination change when in RoutePreview.
+ */
+internal class RouteManager(
+    private val context: DropInNavigationViewContext
+) : UIComponent() {
+
+    private val viewModel = context.viewModel
+    private val fetchAndSetRouteUseCase get() = context.fetchAndSetRouteUseCase()
+
+    private val enabledInStates = listOf(NavigationState.RoutePreview)
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+
+        viewModel.destination.observe { destination ->
+            val navState = viewModel.navigationState.value
+            if (enabledInStates.contains(navState) && destination != null) {
+                fetchAndSetRouteUseCase(destination.point)
+            }
+        }
```

Comment:
This logic competes with the route request that happens inside of `InfoPanelFreeDriveComponent`. It'd be good to think about this redundancy and settle on an approach for route management.

Reply:
That's not a redundancy. 
The RouteManager fetches and sets new route on destination change only when in NavigationState.RoutePreview. 
`InfoPanelFreeDriveComponent` however will do it as a result of a direct user interaction.

Reply:
That's a surprising separation but I think nicely highlights the fact that we discussed yesterday - a need to clean this up and funnel the actions through one unit.

Reply:
Understood. How would you like me to change this? Either class or a sequence UML diagram would be very helpful here.